### DISCUSSION TITLE
Question of Transmission shadow

### DISCUSSION THREAD

**Discussion by llllllleo27** [dreamworksanimation/openmoonray#117] [Q&A] [answered]
Hi，I'm using Moonray v1.3 with Houdini 19.5.640 in Rocky 9.2.
In my test render that shadow of colored glass is just black.This‘s what I got：
![glass shadow](https://github.com/dreamworksanimation/openmoonray/assets/103235311/19bff354-00f0-4124-b99a-c777398471c5)

But in Documentation, I found this
![docs glass](https://github.com/dreamworksanimation/openmoonray/assets/103235311/035423f5-5066-4fb7-8f5d-4d54983f8570)
This should be correct, right?

As same time it shows warning and error:
```
Warning (lib.render): /materials/DwaRefractiveMaterial1/DwaRefractiveMaterial1.iridescence_interpolations: cannot convert TfToken to IntVector
Error: /materials/DwaRefractiveMaterial1/DwaRefractiveMaterial1.iridescence_interpolations: cannot convert TfToken to IntVector
```
I guess this could be the reason？And wonder how to fix that.
If not,what settings should I change to adjust shadows？（I have try some，but not work）

Look forward to your advice.

**Comment by kjeffery** [comment, marked as answer]
Caustics are off by default to reduce noise. You will need to enable caustics. Look for `casts_caustics` in your material.

Another option is to use presence instead of transmission, but it's not physically plausible.

**Reply by llllllleo27** [reply]
Thank you for your reply. And I want to know if it doesn't matter？

Warning (lib.render): /materials/DwaRefractiveMaterial1/DwaRefractiveMaterial1.iridescence_interpolations: cannot convert TfToken to IntVector
Error: /materials/DwaRefractiveMaterial1/DwaRefractiveMaterial1.iridescence_interpolations: cannot convert TfToken to IntVector

**Reply by kjeffery** [reply]
Since you're in Houdini, is it fair to assume you're using the Hydra interface?

We tried to reproduce your error with a DwaRefractiveMaterial with iridescence turned on but were unsuccessful.

**Reply by llllllleo27** [reply]
ok, that's strange.
1. I didn't turn on the iridescence when the error is displayed.
2. After turned on the iridescence it still work even it has an error.

I think maybe something wrong during the compilation process, I'll check.
Thank you for your help.
